output_from_kcl.py

```diff
@@ -0,0 +1,161 @@
+import asyncio
+import os
+import re
+import tomllib
+from concurrent.futures import ProcessPoolExecutor
+from io import BytesIO
+from pathlib import Path
+from typing import Container
+
+import kcl
+import requests
+from kcl import UnitLength
+from PIL import Image
+
+RETRIES = 5
+
+# Map strings to kcl units
+UNIT_MAP = {
+    'in': kcl.UnitLength.In,
+    'mm': kcl.UnitLength.Mm,
+    'ft': kcl.UnitLength.Ft,
+    'm': kcl.UnitLength.M,
+    'cm': kcl.UnitLength.Cm,
+    'yd': kcl.UnitLength.Yd,
+}
+
+
+def export_step(code: str, save_path: Path, unit_length: UnitLength = kcl.UnitLength.Mm) -> bool:
+    try:
+        export_response = asyncio.run(
+            kcl.execute_and_export(code, unit_length, kcl.FileExportFormat.Step)
+        )
+
+        stl_path = save_path.with_suffix(".step")
+
+        with open(stl_path, "wb") as out:
+            out.write(bytes(export_response[0].contents))
+
+        return True
+    except Exception as e:
+        print(e)
+        return False
+
+
+def find_files(
+        path: str | Path, valid_suffixes: Container[str], name_pattern: str | None = None
+) -> list[Path]:
+    """
+    Recursively find files in a folder by a list of provided suffixes or file naming pattern
+
+    Args:
+        path: str | Path
+            Root folder to search
+        valid_suffixes: Container[str]
+            List of valid suffixes to find files by (e.g. ".stp", ".step")
+        name_pattern: str
+            Name pattern to additionally filter files by (e.g. "_component")
+
+    Returns:
+        list[Path]
+    """
+    path = Path(path)
+    files = path.rglob("*")
+    file_paths = []
+    valid_suffixes = [i.lower() for i in valid_suffixes]
+    for file in files:
+        if file.suffix.lower() in valid_suffixes:
+            if name_pattern is not None:
+                if re.match(name_pattern, file.name) is not None:
+                    file_paths.append(file)
+            else:
+                file_paths.append(file)
+    return sorted(file_paths)
```

Comment:
Usually when I see something like `file_paths = []` there is a way to rewrite as a list comprehension.

```suggestion
    path = Path(path)
    valid_suffixes = [i.lower() for i in valid_suffixes]
    return sorted(
        file for file in path.rglob("*")
        if file.suffix.lower() in valid_suffixes and
        (name_pattern is None or re.match(name_pattern, file.name))
    )
```